src/App.js

```diff
@@ -8,46 +8,101 @@ import AddTodoForm from './addTodoForm';
-       setTodoList(result.data.todoList)
-       setIsLoading(false)
-     })
-  }, [])
+  // let initialList = JSON.parse(localStorage.getItem('savedTodoList')) ?? []
+  const [isloading, setIsLoading] = useState(true)
 
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: { 'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}` }
+    }
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(url, options)
 
-  useEffect(() => {
-    if (!isloading) {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList))
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map((todo) => {
+        return { title: todo.fields.Title, id: todo.id, createdAt: todo.createdTime }
+      })
+      const newTodos = todos.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
+      setTodoList(newTodos)
+      setIsLoading(false)
+
+    } catch (error) {
+      console.log(error.message)
     }
-  }, [todoList])
+  }
+
+  const addTodo = async (url, data) => {
+    try {
+      const response = await fetch(url,
+        {
+          method: 'POST',
+          headers: {
+            'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({ records: [{ fields: data }] })
+        });
+
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const newData = await response.json();
+      const listItem = newData.records[0]
+      setTodoList(prevItems => [...prevItems, {
+        title: listItem.fields.Title,
+        id: listItem.id,
 
-  
-  function addTodo(newTodo) {
-    setTodoList(prevItems => [...prevItems, newTodo])
+      }])
+    } catch (error) {
+      console.error('Error updating todo:', error.message);
+      return null
+    }
   }
 
-  function removeTodo(id){
-    const newList =  todoList.filter(item=>
+  const removeTodo = async (id) => {
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`
+    try {
+      const response = await fetch(`${url}/${id}`, {
+        method: 'DELETE',
+        headers: {
+          'Authorization': `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        },
+      });
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      const newList = todoList.filter(item =>
         item.id !== id
       )
       setTodoList(newList)
+    } catch (error) {
+      console.error('Error deleting todo:', error.message);
+
     }
+  }
+
+  useEffect(() => {
+    fetchData()
+  }, [])
+
+  useEffect(() => {
+    if (todoList.length > 0) {
```

Comment:
Nitpick:
This if statement can be simplified to checking for a `0 (false)` and `1+ (true)` length:
```js
if (todoList.length) { ... }
```

Reply:
For sure.

Reply:
<label htmlFor={htmlFor}>{children}:</label>
            <input name={name} id={id} value={value} onChange={onChange} ref={inputRef} />